compiler/rustc_typeck/src/check/fn_ctxt/checks.rs

```diff
@@ -259,6 +264,30 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 return Compatibility::Incompatible(coerce_error);
             }
 
+            // Check that second and third argument of `const_eval_select` must be `FnDef`.
+            //
+            // This is here because there is currently no way to express a trait bound for `FnDef` types only.
```

Comment:
Also the first argument needs to be a tuple, right?

Reply:
Also the comment doesn't mention that the 2nd argument is additionally checked for `const`.

Reply:
added explanation.